simpeg/regularization/pgi.py

```diff
@@ -606,14 +597,12 @@ def deriv(self, m):
 
                 else:
                     r0 = (self.W * (mkvc(dm))).reshape(dm.shape, order="F")
-                    r = mkvc(
-                        np.r_[
-                            [
-                                np.dot(self.gmm.precisions_[membership[i]], r0[i])
-                                for i in range(len(r0))
-                            ]
-                        ]
-                    )
+                    r = np.zeros_like(r0)
+                    for i in range(self.gmm.n_components):
+                        r[membership == i] = (
+                            self.gmm.precisions_[i] @ r0[membership == i].T
+                        ).T
```

Comment:
Similar improvement here:
```suggestion
                        selection = membership == i
                        r[selection] = (
                            self.gmm.precisions_[i] @ r0[selection].T
                        ).T
```